ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/exceptions.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from asyncio.exceptions import TimeoutError
+
+try:
+    from grpclib.const import Status
+    from grpclib.exceptions import GRPCError
+except ImportError:
+    HAS_GRPCLIB = False
+else:
+    HAS_GRPCLIB = True
+
+
+def get_cv_client_exception(exception: Exception, cv_client_details: str | None = None) -> Exception or None:
+    """
+    Convert GRPCError or TimeoutError instances to an instance of the relevant subclass of CVClientException.
+
+    Parameters:
+        exception: Exception to convert.
+
+    Returns:
+        None if If the exception is unmatched, otherwise an instance of the relevant CVClientException subclass.
+    """
+    if not HAS_GRPCLIB:
+        raise RuntimeError("Missing Python library 'grpclib'")
+
+    if isinstance(exception, GRPCError):
+        status, message, details = exception.args
+        if status == Status.NOT_FOUND:
+            return CVResourceNotFound(cv_client_details, *exception.args)
+        if status == Status.CANCELLED:
+            return CVTimeoutError(cv_client_details, *exception.args)
+    if isinstance(exception, TimeoutError):
+        return CVTimeoutError(cv_client_details, *exception.args)
+
+    # Last resort return None so calling exception handling can just raise the single error instead of a chain.
+    return None
```

Comment:
# Last resort return None so calling exception handling can just raise the single error instead of a chain.
If we return e the linters ask us to `raise get_cv_client_exception(...) from e` which would lead to raise e from e.